src/util/MatrixUtil.js

```diff
@@ -1,5 +1,97 @@
 L.MatrixUtil = {
 
+	matrixArrayToCssMatrix: function (array) {
+		var is3d = L.Browser.webkit3d || L.Browser.gecko3d || L.Browser.ie3d,
+			str = is3d ? "matrix3d(" + array.join(',') + ")" : "";
+
+		if (!is3d) {
+		  console.log("Your browser must support 3D CSS transforms in order to use DistortableImageOverlay.");
+        }
+
+  		return str;
+	},
+
+	/* Since matrix3d takes a 4*4 matrix, we add in an empty row and column, which act as the identity on the z-axis. */
+	from2dTo3dMatrix: function(m) {
+	 return [
+        m[0], m[3], 0, m[6],
+        m[1], m[4], 0, m[7],
+        0, 0, 1, 0,
+        m[2], m[5], 0, m[8]
+	  ];
+  	},
+
+	multiplyMatrices: function (a, b) {
+  
+		// TODO - Simplify for explanation
```

Comment:
Whoa.... are we able to just include that matrix library?

This is a really big refactor! I kind of think we should do a short-term fix for the scale issue because it's really critical, even if it's just completely removing the limit for now, because if we merge this we could potentially introduce new issues just because of how major it is! What do you think?

Reply:
Yes! These methods are from an MDN tutorial - they are easy to follow but some of them are not performance optimized - MDN recommended `gl-matrix` library. Just trying to get a sense of how everything works 1st.